src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/ManagedSni/SniError.netcore.cs

```diff
@@ -40,7 +42,23 @@ public SniError(SniProviders provider, uint sniErrorCode, Exception sniException
             lineNumber = 0;
             function = string.Empty;
             this.provider = provider;
-            nativeError = nativeErrorCode;
+            if (nativeErrorCode == 0)
+            {
+                nativeError = nativeErrorCode;
```

Comment:
I guess this could still be done always before the if so the else isn't needed.

Reply:
Done